kotlin/src/LogBeat.kt

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018 Sourcerer Inc. All Rights Reserved.
+// Author: Alexander Surkov (alex@sourcerer.io)
+
+package sourcererio.vacuumanalytics
+
+import com.github.kittinunf.fuel.core.FuelError
+import com.github.kittinunf.fuel.core.FuelManager
+import com.github.kittinunf.fuel.core.Method
+import com.github.kittinunf.fuel.core.Request
+import com.github.kittinunf.fuel.core.Response
+import java.net.HttpCookie
+
+/**
+ * Elastic LogStash logging.
+ */
+class LogBeat {
+    private var session: String = ""
+    private val fuelManager = FuelManager()
+
+    /**
+     * Creates a LogBeat instance.
+     *
+     * @param path [in] path to send logs to,
+     * @param res  [in] HTTP response of successful user authorization,
+     * @param sid  [in] session id of the HTTP response.
+     */
+    constructor(path: String, res: Response, sid: String) {
+        res.headers["Set-Cookie"]?.
+            flatMap { HttpCookie.parse(it) }?.
+            find { it.name == sid }?.
+            let { session = it.value }
+
+        fuelManager.basePath = path
+    }
+
+    /**
+     * Logs error.
+     */
+    fun error(message: String) {
+        beat("error", message)
+    }
+
+    /**
+     * Logs a beat.
+     */
+    fun beat(type: String, message: String) {
```

Comment:
Can we call it event? Beat is very unusual.

Reply:
I was borrowing 'beat' from elastic vocabulary, they postfix all things, that collect data, by 'Beat', like FileBeat and MetricBeat. If we target to elastic primarily, then 'beat' might be a good choice.